api/app/controllers/folders_controller.rb

```diff
@@ -0,0 +1,66 @@
+class FoldersController < ApplicationController
+  # before_action :authenticate_user!, except: [:index, :show]
+  before_action :authorize_user!, only: [ :update, :destroy ]
+
+
+  # GET /folders
+  def index
+    folders = Folder.all
+    render json: FolderSerializer.new(folders).serializable_hash.to_json
+  end
+
+  # GET /my_folders
+  def my_folders
+    authenticate_user!
+    folders = current_user.folders
+    render json: FolderSerializer.new(folders).serializable_hash.to_json
+  end
+
+  # GET /folders/:id
+  def show
+    render json: FolderSerializer.new(folder).serializable_hash.to_json
+  end
+
+  # POST /folders
+  def create
+    folder = Folder.new(folder_params)
+    folder.user = current_user if user_signed_in?
```

Comment:
Is folder.user being nil if the user isnt signed in what you want? Could totally be the case but I wanted to call it out